src/python_vali/src/PySurface.cpp

```diff
@@ -162,6 +162,52 @@ void Init_PySurface(py::module& m) {
 
   py::class_<Surface, shared_ptr<Surface>>(
       m, "Surface", "Image stored in vRAM. Consists of 1+ SurfacePlane(s).")
+      .def(py::init([](Pixel_Format format, uint32_t width, uint32_t height, int gpuID) {
```

Comment:
Please consider taking a `py::object` which has `Width, Height, Format` attributes.
It will allow to construct a Surface not only from `PyDecoder`, but e. g. from `PyNvEncoder` as well.